luminoso_api/v5_client.py

```diff
@@ -176,33 +178,39 @@ def connect_with_username_and_password(cls, url=None, username=None,
             url = URL_BASE + '/' + url.lstrip('/')
             root_url = URL_BASE
 
-        return cls(v4client.session, root_url)
+        return cls(v4client.session, root_url, timeout=timeout)
 
     def _request(self, req_type, url, **kwargs):
         """
         Make a request via the `requests` module. If the result has an HTTP
         error status, convert that to a Python exception.
         """
         kwargs.setdefault('headers', {})['user-agent'] = self.user_agent
+        if self.timeout is not None:
+            kwargs['timeout'] = self.timeout
         logger.debug('%s %s' % (req_type, url))
-        result = self.session.request(req_type, url, **kwargs)
         try:
-            result.raise_for_status()
-        except requests.HTTPError:
-            error = result.text
+            result = self.session.request(req_type, url, **kwargs)
             try:
-                error = json.loads(error)
-            except ValueError:
-                pass
-            if result.status_code in (401, 403):
-                error_class = LuminosoAuthError
-            elif result.status_code in (400, 404, 405):
-                error_class = LuminosoClientError
-            elif result.status_code >= 500:
-                error_class = LuminosoServerError
-            else:
-                error_class = LuminosoError
-            raise error_class(error)
+                result.raise_for_status()
+            except requests.HTTPError:
+                error = result.text
+                try:
+                    error = json.loads(error)
+                except ValueError:
+                    pass
+                if result.status_code in (401, 403):
+                    error_class = LuminosoAuthError
+                elif result.status_code in (400, 404, 405):
+                    error_class = LuminosoClientError
+                elif result.status_code >= 500:
+                    error_class = LuminosoServerError
+                else:
+                    error_class = LuminosoError
+                raise error_class(error)
+        except (requests.Timeout, requests.exceptions.ConnectTimeout,
```

Comment:
ConnectTimeout and ReadTimeout are subclasses of Timeout; we don't need to catch them separately here.

Reply:
Fixed